x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/ExecuteStepsUpdateTask.java

```diff
@@ -86,12 +86,11 @@ public ClusterState execute(final ClusterState currentState) throws IOException
             // either get to a step that isn't a cluster state step or a
             // cluster state wait step returns not completed
             while (currentStep instanceof ClusterStateActionStep || currentStep instanceof ClusterStateWaitStep) {
-                nextStepKey = currentStep.getNextStepKey();
                 if (currentStep instanceof ClusterStateActionStep) {
                     // cluster state action step so do the action and
                     // move the cluster state to the next step
-                    logger.trace("[{}] performing cluster state action ({}) [{}], next: [{}]",
-                        index.getName(), currentStep.getClass().getSimpleName(), currentStep.getKey(), currentStep.getNextStepKey());
+                    logger.trace("[{}] performing cluster state action ({}) [{}]",
+                        index.getName(), currentStep.getClass().getSimpleName(), currentStep.getKey());
                     try {
                         state = ((ClusterStateActionStep) currentStep).performAction(index, state);
```

Comment:
In the future perhaps we will want to return `Tuple<ClusterState, Step.SteyKey>` from this. The disconnect in terms of the ordering of how things have to be called feels a little weird to me, I'd prefer it to be completely functional.

What do you think?

Reply:
I think that would be a reasonable direction to go, yes.